### PR TITLE
chore: add documentations for field access properties

### PR DESCRIPTION
**Pull request by WilsonLe**
## Description

Add documentations for field access properties. This is a quality of life improvements for developers that works on field level access control. Instead of alt+tabbing to payload docs, developers can now read the description of each FieldAccess property. It gets better in vscode where hovering the property displays the documentation.

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] Chore (non-breaking change which does not add functionality)

## Checklist:

- [x] Existing test suite passes locally with my changes

### CODE REVIEW

**Comment by jacobsfletch** [comment]
Hey @WilsonLe I see what you're getting at and I think this would be a good improvement if we can clean up the wording a bit. We need to be able to write these descriptions in as few words as possible and at first pass it was pretty difficult for me to decipher the notes. But I think this will be easy. Here's an example:

Instead of:

```ts
  /**
   * For Collection Create Operation: The full data passed to create the document.\
   * For Collection Read Operation: undefined\
   * For Collection Update Operation: The full data passed to update the document.\
   * For Global Read: undefined\
   * For Global Update: The data passed to update the global with.
   */
  data?: Partial<T>
```

Do this:

```ts
  /**
   * Data is `undefined` during `read` operations
   */
  data?: Partial<T>
```

And instead of this:

```ts
  /**
   * For Collection Create Operation: undefined\
   * For Collection Read Operation: `id` of the document being read\
   * For Collection Update Operation: `id` of the document being updated\
   * For Global Read: undefined\
   * For Global Update: undefined\
   */
  id?: number | string
```

Do this:

```ts
  /**
   * The `id` is only present for existing documents within a collection after they have been created
   */
  id?: number | string
```

And so on. What do you think?

**Comment by WilsonLe** [comment]
Agreed with the recommendations. Pushed new changes. Let me know if you think it's a great balance between concise and informative.

File: packages/payload/src/fields/config/types.ts [hunk 1]
```diff
@@ -50,10 +50,23 @@ export type FieldHook<T extends TypeWithID = any, P = any, S = any> = (
 ) => P | Promise<P>
 
 export type FieldAccess<T extends TypeWithID = any, P = any, U = any> = (args: {
+  /**
+   * The full data passed to `create` or `update` the document. `data` is undefined during `read` operation.
```
Comment (jacobsfletch) [inline packages/payload/src/fields/config/types.ts:54]:
```suggestion
   * The incoming data used to `create` or `update` the document with. `data` is undefined during the `read` operation.
```

File: packages/payload/src/fields/config/types.ts [hunk 2]
```diff
@@ -50,10 +50,23 @@ export type FieldHook<T extends TypeWithID = any, P = any, S = any> = (
...
+   */
   data?: Partial<T>
+  /**
+   * The full document data before `update` is applied. `doc` is undefined during `create` and `update` operation.
```
Comment (jacobsfletch) [inline packages/payload/src/fields/config/types.ts:58]:
```suggestion
   * The original data of the document before the `update` is applied. `doc` is undefined during the `create` operation.
```

File: packages/payload/src/fields/config/types.ts [hunk 3]
```diff
@@ -50,10 +50,23 @@ export type FieldHook<T extends TypeWithID = any, P = any, S = any> = (
...
+   */
   doc?: T
+  /**
+   * `id` of the document being `read` or `update`. `id` is undefined during `create` operation.
```
Comment (jacobsfletch) [inline packages/payload/src/fields/config/types.ts:62]:
```suggestion
   * The `id` of the current document being read or updated. `id` is undefined during the `create` operation.
```

File: packages/payload/src/fields/config/types.ts [hunk 4]
```diff
@@ -50,10 +50,23 @@ export type FieldHook<T extends TypeWithID = any, P = any, S = any> = (
...
+   */
   id?: number | string
+  /** The `Express` request object containing the currently authenticated `user` */
   req: PayloadRequest<U>
+  /**
+   * Immediately adjacent field data passed to `create`, `read`, and `update` document with.
```
Comment (jacobsfletch) [inline packages/payload/src/fields/config/types.ts:68]:
```suggestion
   * Immediately adjacent data to this field. For example, if this is a `group` field, then `siblingData` will be the other fields within the group.
```

**Comment by jacobsfletch** [comment]
I think we're almost there @WilsonLe, I left a couple suggestions.

**Comment by WilsonLe** [comment]
@jacobsfletch I added your suggestions in the latest commit. The wording looks better now.

**Pull request merged**